### PR TITLE
Allow the tagName to be a custom component

### PR DESCRIPTION
**Pull request by aij**
It already worked as expected but was producing PropType
warnings in development mode.

### CODE REVIEW

**Comment by jacobrask** [comment]
It should probably be named something else in that case, perhaps `component`.

**Comment by aij** [comment]
I agree, but renaming it would break backwards compatibility.

I guess the other option would be to allow both names, and have one take precedence over the other, but that also seems a bit goofy.

I'm fine with either. Just let me know what you'd like.

File: src/GatewayDest.js [hunk 1]
```diff
@@ -8,7 +8,11 @@ export default class GatewayDest extends React.Component {
 
   static propTypes = {
     name: React.PropTypes.string.isRequired,
-    tagName: React.PropTypes.string
+    tagName: React.PropTypes.oneOfType([
+      React.PropTypes.string,
+      React.PropTypes.func,
+      React.PropTypes.instanceOf(React.Component)
```
Comment (thejameskyle) [inline src/GatewayDest.js]: I think it only needs `React.PropTypes.func`: http://react.jsbin.com/jiluza/edit?js,console

Comment (aij) [inline src/GatewayDest.js]: You're right. Removed.

**Pull request merged**